src/System.Private.CoreLib/src/System/Threading/LowLevelLifoSemaphore.cs

```diff
@@ -0,0 +1,285 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace System.Threading
+{
+    /// <summary>
+    /// A LIFO semaphore.
+    /// Waits on this semaphore are uninterruptible.
+    /// </summary>
+    //[StructLayout(LayoutKind.Sequential)] // enforce layout so that padding reduces false sharing
+    internal partial class LowLevelLifoSemaphore : IDisposable
+    {
+        //private Internal.PaddingFor32 pad1;
+        private Counts _counts;
+        //private Internal.PaddingFor32 pad2;
```

Comment:
I forgot to re-enable this. I'll likely need to wrap it in a `struct` inside the `class` since partial classes cannot use sequential layout.